src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,276 @@
+                throw new IllegalArgumentException(
+                    "parameters for input field [" + inputField + "] cannot be cast to [" + Map.class.getName() + "]"
+                );
+            }
+
+            Map<String, Object> parameterMap = (Map<String, Object>) parameters;
+
+            // output field must be string
+            if (!(parameterMap.containsKey(OUTPUT_FIELD))) {
+                throw new IllegalArgumentException(
+                    "parameters for input field [" + inputField + "] misses [" + OUTPUT_FIELD + "], cannot process it."
+                );
+            }
+
+            Object outputField = parameterMap.get(OUTPUT_FIELD);
+
+            if (!(outputField instanceof String)) {
+                throw new IllegalArgumentException(
+                    "parameters for output field [" + OUTPUT_FIELD + "] cannot be cast to [" + String.class.getName() + "]"
+                );
+            }
+
+            // check non string parameters
+            int chunkingAlgorithmCount = 0;
+            Map<String, Object> chunkerParameters;
+            for (Map.Entry<?, ?> parameterEntry : parameterMap.entrySet()) {
+                if (!(parameterEntry.getKey() instanceof String)) {
+                    throw new IllegalArgumentException("found parameter entry with non-string key");
+                }
+                String parameterKey = (String) parameterEntry.getKey();
+                if (supportedChunkers.contains(parameterKey)) {
+                    chunkingAlgorithmCount += 1;
+                    chunkerParameters = (Map<String, Object>) parameterEntry.getValue();
+                    IFieldChunker chunker = ChunkerFactory.create(parameterKey, analysisRegistry);
+                    chunker.validateParameters(chunkerParameters);
+                }
+            }
+
+            // should only define one algorithm
+            if (chunkingAlgorithmCount != 1) {
+                throw new IllegalArgumentException("input field [" + inputField + "] should has and only has 1 chunking algorithm");
+            }
+        }
+    }
+
+    private void validateContent(Object content, String inputField) {
+        // content can be a map, a list of strings or a list
+        if (content instanceof Map) {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> contentMap = (Map<String, Object>) content;
+            for (Map.Entry<String, Object> contentEntry : contentMap.entrySet()) {
+                String contentKey = contentEntry.getKey();
+                Object contentValue = contentEntry.getValue();
+                // the map value can also be a map, list or string
+                validateContent(contentValue, inputField + "." + contentKey);
+            }
+        } else if (content instanceof List) {
+            List<?> contentList = (List<?>) content;
+            for (Object contentElement : contentList) {
+                if (!(contentElement instanceof String)) {
```

Comment:
We should support list or map inside a list.

Reply:
Do not worry. Through our discussion today, the chunking processor will extend and follow the inference processor.